Implemented the Line.project (#3402)
pygame-community/pygame-ce: docs/reST/ref/geometry.rst

         .. versionadded:: 2.5.3

      .. ## Line.flip_ab_ip ##

   .. method:: project

         | :sl:`projects the line onto the given line`
         | :sg:`project(point: tuple[float, float], clamp=False) -> tuple[float, float]`

         This method takes in a point and one boolean keyword argument clamp. It outputs an orthogonally projected point onto the line.

>          If clamp is True it makes sure that the outputted point will be on the line segment.

Reviewer: ```suggestion
         If clamp is `True` it makes sure that the outputted point will be on the line segment (which might not be orthogonal), and if it is `False` (the default) then any point on the infinitely extended line may be outputted.
         This method can be used to find the closest point on a line to the given point. The output is the unique point on the line or line segment that is the smallest distance away from the given point.
```
- Clarify `clamp` argument more
- Add intuitive definition (biggest use case)

         .. figure:: code_examples/project.png
            :alt: project method image

            Example of how it projects the point on to the line purplish point is the point we want to project and the blue point is what you would get as a result

         .. figure:: code_examples/project_clamp.png

>             :alt: project do_clamp=True image
> 
>             Example of what the clamp argument changes if it is true it stays on the line segment

Reviewer: ```suggestion
            :alt: project clamp argument image

            Example of what the clamp argument changes. If it is `True`, the point is bounded between the line segment ends.
```

         This method takes in a point and one boolean keyword argument clamp. It outputs an orthogonally projected point onto the line.
         If clamp is `True` it makes sure that the outputted point will be on the line segment (which might not be orthogonal), and if it is `False` (the default) then any point on the infinitely extended line may be outputted.
         This method can be used to find the closest point on a line to the given point. The output is the unique point on the line or line segment that is the smallest distance away from the given point.

         .. figure:: code_examples/project.png
            :alt: project method image

            Example of how it projects the point onto the line. The red point is the point we want to project and the blue point is what you would get as a result.

         .. figure:: code_examples/project_clamp.png
            :alt: project clamp argument image

            Example of what the clamp argument changes. If it is `True`, the point is bounded between the line segment ends.

            WARNING: This method has to have some length or the clamp parameter must be true for it to work and not throw a `ValueError`

>          .. versionadded:: 2.5.4

Reviewer 2: this would need to be updated to 2.5.6 after we have consensus on this PR

         .. versionadded:: 2.5.3

      .. ## Line.flip_ab_ip ##

   .. method:: project

         | :sl:`projects the line onto the given line`
         | :sg:`project(point, do_clamp=False) -> point`

>          Returns a new point(tuple[float, float]) that is projected onto the line like so

Reviewer: I think the docs are lacking for this new method. The examples are good, but they cannot be the only description of the behavior.
- Add a technical definition involving orthagonality or something.
- Add an intuitive definition: The result is the *closest* point on the extended line (or line segment if `do_clamp` is true) to the given point.
- In the signature, replace the return type `point` with `tuple[float, float]`, and then remove `(tuple[float, float])` from the description.

         .. figure:: code_examples/project1.png
            :alt: project method image

>             Example of what project method does if you input the green point you get back the yellow one.

Reviewer: Confusing grammar.
```suggestion
            Example of what project method does.
            Green point: input argument. Yellow point: return value.
```
Additionally, maybe the yellow point should be purple instead for consistency with the other examples.
